### ISSUE TITLE
ResolveAll<T> not throwing exception

### ISSUE THREAD

**Issue by mack0196** [unitycontainer/abstractions#56]
This maybe a clarification of expected behavior, but I have a scenario where I have a registration done with a name-key. When I Resolve<T>(key) I get an exception; this is expected because config that ctor depends on is bad. When I ResolveAll<T> I get a T[0] and no exception.

There is only that single registration done so I expect 1 when ResolveAll<T> works or an exception when ResolveAll<T> fails.

In short this created an exception-less fail state when my app starts up.

**Comment by mack0196** [comment]
Any feedback on this behavior and potential change?

**Comment by ENikS** [comment]
This behavior is by design.

**Comment by mack0196** [comment]
Is there a way to get these failures to come up the stack?
